cli/impl/src/main/java/io/helidon/build/cli/impl/HelidonVersion.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.build.cli.impl;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.helidon.build.common.maven.MavenVersion;
+import io.helidon.build.common.maven.VersionRange;
+
+/**
+ * Support Maven version ranges for archetypes.
+ */
+class HelidonVersion {
```

Comment:
How about using another approach.

E.g. adding a new static method to `VersionRange`

```java
public static VersionRange wildcard(String version) {
    if (version.endsWith("-latest") || version.endsWith("-LATEST")) {
        return createFromVersionSpec("[" + version.substring(0, version.length() - 7) + ",)");
    } else {
        return createFromVersionSpec("[" + version + "]");
    }
}
```

The version can be resolved

```java
private MavenVersion resolve(String helidonVersion) throws Metadata.UpdateFailed {
    ArchetypesData data = metadata().archetypesData();
    return VersionRange.wildcard(helidonVersion).matchVersion(data.mavenVersions());
}
```